### DISCUSSION TITLE
If else statement

### DISCUSSION THREAD

**Discussion by pkleanthous** [go-task/task#429] [Q&A]
Hey,

I'm using the task runner daily. It's a great tool, thanks for inventing it :)
I'm wondering how can I implement if-else statements and run different tasks.

If you the tool has the above functionality, can you provide an example?

**Comment by andreynering** [comment]
Hi @pkleanthous,

Sorry, but we do not have a way to do if/else statements, unless *inside* Bash commands.

Do you have an example of what you're trying to accomplish?

**Reply by pkleanthous** [reply]
Hi @andreynering,

I'm using go-task daily and I would like eventually to replace all my bash scripts and makefiles.
Regularly I'm using if/else statemtents either to execute different stuff based on a variable or based on a command output.

An example can be:

```
if [ "${workspace_missing}" = "0" ]; then 
  terraform workspace new ${workspace}
else 
  terraform workspace select ${workspace}
fi
```
OR

```
if [[ "${env}" =~ (prod) ]]; then
  if [[ "$*" =~ apply ]]; then
    terraform $*
  else
    terraform $* ${cd_vars}
  fi
else
  terraform $* ${cd_vars}
fi
```

**Reply by andreynering** [reply]
Unfortunately we don't have a way to accomplish that using Task alone yet. Only using Bash, as I said.

I understand such feature would be useful, but I don't have a good UX in mind at the moment, nor I have certainty that I really want that feature to be implemented at the moment.

**Comment by SebSa** [comment]
Any update on this?

**Reply by ghostsquad** [reply]
I think the thread speaks for itself. This is a interesting use case though, and it fits well with what I'm working on with fngo.

**Comment by ansidev** [comment]
@andreynering For the simple if-else statement

```shell
if [ condition ]; then 
  command
else 
  error_handler
fi
```

Here is my idea:

```yaml
task_name:
  cmds:
    - if_else:
      - condition: condition
      - on_true: command
      - on_false: error_handler
    // other commands
```

`condition`, `command`, and `error_handler` can be shell commands or other task names.

**Comment by andreynering** [comment]
For those landing into this discussion, we now have issues for these feature requests:

- https://github.com/go-task/task/issues/608
- https://github.com/go-task/task/issues/978
